include/oneapi/tbb/detail/_template_helpers.h

```diff
@@ -182,7 +182,17 @@ using pack_element_t = typename pack_element<N, Args...>::type;
 template <typename Func>
 class raii_guard {
 public:
-    raii_guard( Func f ) : my_func(f), is_active(true) {}
+    static_assert(
+        std::is_nothrow_copy_constructible<Func>::value &&
+        std::is_nothrow_move_constructible<Func>::value,
+        "Throwing an exception during the Func copy or move construction cause an unexpected behavior."
+    );
+
+    raii_guard( Func f ) noexcept : my_func(f), is_active(true) {}
```

Comment:
Why not add an overload accepting functor via rvalue ref ?

Reply:
In this part of the functionality (under this I mean `raii_guard`, `try_call`, `on_exception`, `on_completion`...), all functors objects are passed by copy to let the compiler better optimize the code. We already looked at this, check the assembler and it looks like compilers inline ALL `try_call()` cases and just call the required functors.

So it was decided not to apply passing by reference here to keep the inlining.

Reply:
I asking not to change existing one method, but instead add a _new_ one

Reply:
But what is the purpose of this overload and why do you think that we need to add such one?